### PR TITLE
Notifications UI

### PR DESCRIPTION
**Pull request by oliviaxjiang**
- added a notifications button at the top right corner of home screen
- added a NotificationsHomeScreen
- added a settings row that allows the user to navigate to NotificationsSettings
- added a NotificationsSettingsScreen

https://github.com/user-attachments/assets/3cb28d6f-9642-4808-8e90-90e844e6aab4

### CODE REVIEW

**Review by zachseidner1 — APPROVED** [review]
PR reviewed in-person, nice job!

File: app/src/main/java/com/cornellappdev/android/eatery/ui/components/notifications/FavoriteItemRow.kt [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+ * - If there are three or more eateries, it displays the condensed eatery name of the last eatery
+ *      followed by "and n other eateries," where n is the count of additional eateries.
+ *
+ * @param eateries A list of eatery names to display.
+ * @Composable
+ * @return A Text composable that shows the primary eatery name with a summary of additional eateries,
+ *      if any.
+ */
+@Composable
+private fun condenseEateriesName(atEateries: List<String>) {
```
Comment (thisjustin123) [inline app/src/main/java/com/cornellappdev/android/eatery/ui/components/notifications/FavoriteItemRow.kt:103]: Capitalize function name :D

File: app/src/main/java/com/cornellappdev/android/eatery/ui/components/notifications/FavoriteItemRow.kt [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    val condensed = atEateries.map { condenseDiningHallNames(it) }
+    val text = if (atEateries.size > 1) {
+        "${condensed.last()} + ${atEateries.size - 1} other"
+    } else {
+        condensed.lastOrNull() ?: ""
+    }
```
Comment (thisjustin123) [inline app/src/main/java/com/cornellappdev/android/eatery/ui/components/notifications/FavoriteItemRow.kt:109]: Love this function tho!! really great stuff, love the documentation too

File: app/src/main/java/com/cornellappdev/android/eatery/ui/screens/NotificationsHomeScreen.kt [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import com.cornellappdev.android.eatery.ui.components.notifications.FavoriteItemRow
+import com.cornellappdev.android.eatery.ui.theme.EateryBlue
+import com.cornellappdev.android.eatery.ui.theme.EateryBlueTypography
+
+@Composable
+fun NotificationsHomeScreen(
+
+) {
```
Comment (thisjustin123) [inline app/src/main/java/com/cornellappdev/android/eatery/ui/screens/NotificationsHomeScreen.kt:18]: Ah ok so this is just UI for this screen, no ViewModel. Add a TODO to add a hilt viewmodel here :-)

File: app/src/main/java/com/cornellappdev/android/eatery/ui/screens/NotificationsSettingsScreen.kt [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        )
+
+        SwitchOption(
+            title = "All Notifications",
+            description = "",
+            initialValue = allNotificationsEnabled,
+            onCheckedChange = { enabled ->
+                allNotificationsEnabled = enabled
+            }
+        )
```
Comment (thisjustin123) [inline app/src/main/java/com/cornellappdev/android/eatery/ui/screens/NotificationsSettingsScreen.kt:57]: Oh weird, this was an existing function? Just one small nitpick, the padding on the switch looks a little sus; it's very close to the edge on the right:
image

vs. Figma
image

Plz make it `16.dp` padding (on both start and end) :-)

**Review by thisjustin123 — APPROVED** [review]
Really awesome stuff, UI looks SUPER clean. Left a few tiny comments to clean up before it is good to merge :-)

**Comment by oliviaxjiang** [comment]
this is now the feature branch for notifications

**Pull request closed without merging**